src/geometry/ArborX_DetailsAlgorithms.hpp

```diff
@@ -112,6 +110,53 @@ KOKKOS_INLINE_FUNCTION decltype(auto) returnCentroid(Geometry const &geometry)
                             Geometry>::apply(geometry);
 }
 
+template <class Geometry, class Space = Kokkos::HostSpace>
```

Comment:
Don't know if `Space` should have a default. `Kokkos::Sum` doesn't but it is used as `Sum<value>` all over the place.